modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientExperimentalCodegen.java

```diff
@@ -910,7 +915,15 @@ public String getTypeString(Schema p, String prefix, String suffix) {
         } else if (ModelUtils.isArraySchema(p)) {
             ArraySchema ap = (ArraySchema) p;
             Schema inner = ap.getItems();
-            return prefix + "[" + getTypeString(inner, "", "") + "]" + fullSuffix;
+            if (inner == null) {
+                // In OAS 3.0.x, the array "items" attribute is required.
+                // In OAS >= 3.1, the array "items" attribute is optional such that the OAS
+                // specification is aligned with the JSON schema specification.
+                // When "items" is not specified, the elements of the array may be anything at all.
+                return "[bool, date, datetime, dict, float, int, list, str, none_type]";
```

Comment:
This is the 3rd location in this function that w have defined this list of all types.
Instead of defining this same info 3 times, how about doing one more recursive call and routing us to the return case on line 903 or 910? That keeps this simpler.

Reply:
I'm not convinced by your argument. I think recursion will make the code more complicated to understand and troubleshoot. Plus here the inner schema is null, so getTypeString() would have to be invoked with either a null value, or create/use a new AnyType schema

Reply:
Yup, my thought was that one would invoke it with the null Schema input.
- line 925 does this same [] wrapping logic
- The returned data is the line 903 anytype data, all we would need to add is a null check in line 902
- The rest of the function uses recursion
But if this is clearer to you we can keep it this way.

Reply:
I have added code comments and used recursion. Thanks for your input, that helps to improve the doc.
I have put some additional code comments for recursion because I think the recursion complicates readability and maintainability.